Source/Core/VideoCommon/VertexLoaderX64.cpp

```diff
@@ -310,6 +321,13 @@ void VertexLoaderX64::GenerateVertexLoader()
 		MOVZX(32, 8, scratch1, MDisp(src_reg, m_src_ofs));
 		AND(32, R(scratch1), Imm8(0x3F));
 		MOV(32, MDisp(dst_reg, m_dst_ofs), R(scratch1));
+		if (m_VtxDesc.PosMatIdx)
+		{
+			CMP(32, R(count_reg), Imm8(3));
```

Comment:
You could make position_cache a circular buffer, and store every vertex to it, I guess?  Not sure.